Formula/tree-sitter.rb

```diff
@@ -0,0 +1,51 @@
+class TreeSitter < Formula
+  desc "Parser generator tool and incremental parsing library"
+  homepage "https://tree-sitter.github.io/"
+  url "https://github.com/tree-sitter/tree-sitter/archive/0.16.5.tar.gz"
+  sha256 "3c6357d1a153c43f481d07f1a8743ed8b3134a2934689d564795a0d083ff7c7e"
+  head "https://github.com/tree-sitter/tree-sitter.git"
+
+  def install
+    # this shell script invokes clang or CC, and builds a .a with ar
+    system "script/build-lib"
+
+    lib.install "libtree-sitter.a"
+    include.install "lib/include/tree_sitter"
+  end
```

Comment:
Shouldn't it just depend on rust and be installed with cargo?

Reply:
This library is C, but does come with Rust bindings. It seems likely that building the bindings also build the C library and maybe even installs it in the same way. This formula just follows the published tree-sitter documentation for a standard C build/install.

Do you think I need to look into building all the bindings too? There are a few different languages supported, but I only have experience with the default C library.

Reply:
Apparently there's also a rust CLI. It might be good to include that one.

Reply:
Yeah, looks like there is a Rust CLI part of the build process, any idea if we want to include that in here too?

Reply:
I haven't ever used it before as part of my work with tree-sitter. But, I'll investigate and report back.